### PR TITLE
WIP: adding a md linter to the lint process

### PR DESCRIPTION
**Pull request by joshuafernandes**

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -12,5 +12,7 @@ max_line_length = 80
 trim_trailing_whitespace = true
 
 [*.md]
+max_line_length = 80
```
Comment (m4sterbunny) [inline .editorconfig:15]:
```suggestion
max_line_length = 100
```

File: .editorconfig [hunk 2]
```diff
@@ -8,9 +8,5 @@ end_of_line = lf
 insert_final_newline = true
 indent_style = space
 indent_size = 2
-max_line_length = 80
+max_line_length = 100
 trim_trailing_whitespace = true
-
-[*.md]
```
Comment (joshuafernandes) [inline .editorconfig:14]: Keep the config in the rules section so its one place

File: .eslintrc.md.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  ],
+  "parser": "eslint-plugin-markdownlint/parser",
+  "parserOptions": {
+    "ecmaVersion": "latest",
+    "sourceType": "module",
+    "ecmaFeatures": {
+      "jsx": true
+    },
+  },
+  rules: {
```
Comment (joshuafernandes) [inline .eslintrc.md.js:13]: @m4sterbunny all the various rules to get things going. Have fixed up the bits in md/mdx as well so it now runs as part of the lint step

File: .github/workflows/spelling.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  vale:
+    name: Spelling
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Vale
+        uses: ConsenSys/docs-gha/spelling@main
```
Comment (joshuafernandes) [inline .github/workflows/spelling.yml:19]: @m4sterbunny If you merge [#6 ](https://github.com/Consensys/docs-gha/pull/6) , or copy pasta the content as a branch directly into `docs-gha` ^ change the above line `main` to your branch's name and you can test things on this repo?

File: .eslintrc.md.js (hunk 3 above)
Comment (m4sterbunny) [reply .eslintrc.md.js:13]: @joaniefromtheblock FYI Josh has setup some of the rules to get ES link going

**Pull request merged**